src/db/Database.ts

```diff
@@ -29,6 +34,21 @@ export class Database<Dictionary extends ModelDictionary> {
       },
       {},
     )
+
+    callOrder++
+    this.id = this.generateId()
+  }
+
+  /**
+   * Generates a unique MD5 hash based on the database
+   * module location and invocation order. Used to reproducibly
+   * identify a database instance among sibling instances.
+   */
+  private generateId() {
+    const { stack } = new Error()
+    const callFrame = stack?.split('\n')[4]
+    const salt = `${callOrder}-${callFrame?.trim()}`
```

Comment:
I'm using an invocation order and database location to identify a single database instance. 
The order of `factory()` calls is unlikely to change on runtime. Two first-created databases across clients aren't necessarily the same instances, as each client may load a different module, while the same `BroadcastChannel` controls _the entire host_. To account for that, I'm adding a module location to the hash salt. 

Two database instances on multiple pages are equal if:
1. They were called in the same order.
2. They were created in the same module.